src/modules/deal/deals.service.ts

```diff
@@ -0,0 +1,181 @@
+import { Injectable } from '@nestjs/common';
+import { DEFAULTS } from '@ukef/constants/properties';
+import { AcbsService } from '@ukef/modules/acbs/acbs.service';
+import { AcbsAuthenticationService } from '@ukef/modules/acbs/acbs-authentication.service';
+
+import { CreateDealDto } from './dto/deals-request.dto';
+
+@Injectable()
+export class DealsService {
+  constructor(private readonly acbsAuthenticationService: AcbsAuthenticationService, private readonly acbsService: AcbsService) {}
+
+  async createDeal(createDealDto: CreateDealDto): Promise<any> {
+    const idToken = await this.acbsAuthenticationService.getIdToken();
+
+    const todayDate = new Date();
+    todayDate.setHours(0, 0, 0, 0); // Set to midnight
```

Comment:
Might be best to convert this to `getMidnightTodayDate` or something along those lines as this will be re-used in other endpoints as well.

Reply:
Ah, nice, also avoids the need for a comment 🙂